File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group = defaultdict(list)

        for word in strs:
            original = word
            word = "".join(sorted(word))
            group[word].append(original)
        
        ans = []
        for key in group:
            ans.append(group[key])
        
        return ans

File: 0049-group-anagrams/0049-group-anagrams.swift
class Solution {
    func groupAnagrams(_ strs: [String]) -> [[String]] {
        var dict: [String: [String]] = [:]
        
        for str in strs {
            let sorted_str = String(str.sorted())
            dict[sorted_str, default: []].append(str)
        }

        return Array(dict.values)
    }
}